cluster/saltbase/salt/docker/docker-healthcheck

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Copyright 2015 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is intended to be run periodically, to check the health
+# of docker.  If it detects a failure, it will restart docker using systemctl.
```

Comment:
Yes that is true - I am less trusting of Docker than you are, so I give it longer just in case it wakes up ;-)

More seriously, because this is an entirely separate service (unlike on GCE where it supervises things), I tried to be more forgiving.  In particular I wanted to be sure Docker had had a chance to start up when we first run.  I can adjust the timings to be shorter but I don't think the risk/reward is there - hopefully Docker won't crash too often, and saving a minute or two on recovery isn't worth it to me vs accidentally killing a Docker that is healthy but slow.

Happy to adjust these if you think differently though!